telegram/ext/picklepersistence.py

```diff
@@ -198,11 +198,11 @@ def get_conversations(self, name):
         return self.conversations.get(name, {}).copy()
 
     def update_conversation(self, name, key, new_state):
-        """Will update the conversations for the given handler and depending on :attr:`on_flush`
-        save the pickle file.
+        """Will update the conversations (if changed) and depending on :attr:`on_flush`
```

Comment:
please revert. That we only update on change is an implementation details imo. I think you also made this change in dictpersistence …